File: hotel-app/src/main/java/hotelico/service/booking/application/customer/domain/infrastructure/security/CustomUserDetails.java
package hotelico.service.booking.application.customer.domain.infrastructure.security; 

import hotelico.service.booking.adapter.out.persistence.customer.model.UserEntity;
import hotelico.service.booking.application.customer.common.enums.UserRoleEnum;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.AuthorityUtils;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.util.StringUtils;

import java.util.Arrays;
import java.util.Collection;
import java.util.List;

/**
 * This service is needed for SPRING-SECURITY
 */
public class CustomUserDetails extends UserEntity implements UserDetails {

    private final List<String> userRoles;

    public CustomUserDetails(UserEntity user, List<String> userRoles) {
        super(user);
        this.userRoles = userRoles;
    }

    @Override
    public String getPassword() {
        return super.getUserPassword();
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        String roles = StringUtils.collectionToCommaDelimitedString(userRoles);
        return AuthorityUtils.commaSeparatedStringToAuthorityList(roles);
    }

    @Override
    public boolean isAccountNonExpired() {
        return true; // TODO: handle real expiration logic
    }

    @Override
    public boolean isAccountNonLocked() {
        return true; // TODO: handle real lock logic
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true; // TODO: handle real credential expiration logic
    }

    @Override
    public boolean isEnabled() {
        return true; // TODO: handle real enable/disable logic
    }

    @Override
    public String getUsername() {
        return super.getUserName();
    }

    public boolean isAdmin() {
        String roles = StringUtils.collectionToCommaDelimitedString(userRoles);
        String[] split = roles.split(",");
        return Arrays.asList(split).contains(UserRoleEnum.ROLE_HOTELICO_ADMIN.getValue());
    }

    public boolean canEdit() {
        String roles = StringUtils.collectionToCommaDelimitedString(userRoles);
        String[] split = roles.split(",");
        List<String> roleList = Arrays.asList(split);
        return roleList.contains(UserRoleEnum.ROLE_HOTELICO_ADMIN.getValue())
                || roleList.contains(UserRoleEnum.ROLE_HOTELICO_EDITOR.getValue());
    }
}

File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/customer/infrastructure/security/CustomUserDetails.kt
package eu.getsoftware.hotelico.customer.infrastructure.security//package eu.getsoftware.hotelico.clients.infrastructure.service.security

import eu.getsoftware.hotelico.customer.domain.User
import eu.getsoftware.hotelico.customer.domain.enums.UserRoleEnum
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.AuthorityUtils
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.util.StringUtils
import java.util.*

/**
 * This service is needed for SPRING-SECURITY
 */

class CustomUserDetails : User, UserDetails
{

    private val userRoles: List<String>

    constructor(user: User, userRoles: List<String>) : super(user) {
        this.userRoles = userRoles
    }

    override fun getPassword(): String {
        return super.getUserPassword()
    }

    override fun getAuthorities(): Collection<GrantedAuthority> {

        val roles = StringUtils.collectionToCommaDelimitedString(userRoles)
        return AuthorityUtils.commaSeparatedStringToAuthorityList(roles)
    }

    override fun isAccountNonExpired(): Boolean {
        // TODO Auto-generated method stub
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }

    override fun getUsername(): String {
        return super.userName
    }

    fun isAdmin(): Boolean {

        val roleStrings = StringUtils.collectionToCommaDelimitedString(userRoles)

        val split = roleStrings.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        return Arrays.asList(*split).contains(UserRoleEnum.ROLE_HOTELICO_ADMIN.value)
    }    

    fun canEdit(): Boolean {

        val roleStrings = StringUtils.collectionToCommaDelimitedString(userRoles)

        val split = roleStrings.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        val asList = Arrays.asList(*split)

        return asList.contains(UserRoleEnum.ROLE_HOTELICO_ADMIN.value) || asList.contains(UserRoleEnum.ROLE_HOTELICO_EDITOR.value)
    }

}
